src/Player.cpp

```diff
@@ -207,30 +208,30 @@ void Player::UseItem() {
     }
 }
 
-void Player::AddToInventory(vector<int> drops) {
+void Player::AddToInventory(const ItemMap_t & drops) {
     // Adds items to inventory and prints out what the player received.
     // Adds items received to total items.
-    arrows += drops.at(0);
-    bombs += drops.at(1);
-    potions += drops.at(2);
-    whetstones += drops.at(3);
-    coins += drops.at(4);
+    arrows += GetInventory(drops, ITEMTYPE::ARROWS);
+    bombs += GetInventory(drops, ITEMTYPE::BOMB);
+    potions += GetInventory(drops, ITEMTYPE::POTION);
+    whetstones += GetInventory(drops, ITEMTYPE::WHETSTONE);
+    coins += GetInventory(drops, ITEMTYPE::COIN);
     // Prints number of items received.
     cout << "You have gained: " << endl;
-    if (drops[0] > 0) {
-        cout << "[" << drops.at(0) << "] arrows" << endl;
+    if (GetInventory(drops, ITEMTYPE::ARROWS) > 0) {
+        cout << "[" << GetInventory(drops, ITEMTYPE::ARROWS) << "] arrows" << endl;
     }
-    if (drops[1] > 0) {
-        cout << "[" << drops.at(1) << "] bombs" << endl;
+    if (GetInventory(drops, ITEMTYPE::BOMB) > 0) {
+        cout << "[" << GetInventory(drops, ITEMTYPE::BOMB) << "] bombs" << endl;
     }
-    if (drops[2] > 0) {
-        cout << "[" << drops.at(2) << "] potions" << endl;
+    if (GetInventory(drops, ITEMTYPE::POTION) > 0) {
+        cout << "[" << GetInventory(drops, ITEMTYPE::POTION) << "] potions" << endl;
     }
-    if (drops[3] > 0) {
-        cout << "[" << drops.at(3) << "] whetstones" << endl;
+    if (GetInventory(drops, ITEMTYPE::WHETSTONE) > 0) {
+        cout << "[" << GetInventory(drops, ITEMTYPE::WHETSTONE) << "] whetstones" << endl;
     }
-    if (drops[4] > 0) {
-        cout << "[" << drops.at(4) << "] coins" << endl;
+    if (GetInventory(drops, ITEMTYPE::COIN) > 0) {
+        cout << "[" << GetInventory(drops, ITEMTYPE::COIN) << "] coins" << endl;
```

Comment:
While I understand that `GetInventory` is supposed to be a wrapper to handle the case where an item isn't present in the map, I think it adding a static method introduces some unneeded complexity here. I'd recommend something like this:

```
cout << "You have gained: " << endl;

if (drops.find(ITEMTYPE::ARROWS) != drops.end()) {
    arrows += drops.at(ITEMTYPE::ARROWS);
    cout << "[" << drops.at(ITEMTYPE::ARROWS) << "] arrows" << endl;
}

if (drops.find(ITEMTYPE::BOMBS) != drops.end()) {
    bombs += drops.at(ITEMTYPE::BOMBS);
    cout << "[" << drops.at(ITEMTYPE::BOMBS) << "] bombs" << endl;
}

// ... etc ...
```

This code will likely be refactored anyway once #107 is implemented, so even if this way is more verbose I think it's better to be clear.

Reply:
@tagniam fixed. please test